### DISCUSSION TITLE
Override middleware for specific routes

### DISCUSSION THREAD

**Discussion by KHIT93** [nuxt/nuxt#16912] [Questions]
I am working on an application where all routes, except for a few, are protected behind a login.
Since that is the case, I created a `middleware/auth.global.js` and handled the logic to detect if the user is logged ind or not

```javascript
export default defineNuxtRouteMiddleware((to, from) => {
    const { $auth } = useNuxtApp()
    if (!$auth().isAuthenticated()) {
        return navigateTo('/login')
    }

    return
})
```

For the few routes that should be accessible for unauthenticated users, I created a `middleware/guest.js` file
```javascript
export default defineNuxtRouteMiddleware((to, from) => {
    const { $auth } = useNuxtApp()
    if ($auth().isAuthenticated()) {
        return navigateTo('/')
    }

    return
})
```

on `pages/login.vue` I have added the `guest` middleware like this:
```javascript
<script setup>
...
definePageMeta({
    middleware: ['guest']
})
...
</script>
```

If I am not logged in, Nuxt returns `Infinite redirect in navigation guard`.
Renaming `middleware/auth.global.js` to `middleware/auth.js`, resolves the issue, but requires me to specifically define the `auth` middlware for every page.

Is there a way for me to define it in a way where if I set the middleware for a page, fit will override the global middleware for just that route or would I need to a ignore-list to the `middleware/auth.global.js` where I skip everything in case of accessing a route that should not be handled by the global middleware?

**Comment by sisou** [comment]
You'll need to add an ignore-list in the global middleware, there is no other way to exclude a global middleware for a route.
